var/spack/repos/builtin/packages/abinit/package.py

```diff
@@ -71,111 +73,101 @@ class Abinit(Package):
         else:
-            oapp("--with-linalg-flavor=custom")
-            linalg = spec['lapack'].libs + spec['blas'].libs
+            oapp('--with-linalg-flavor=custom')
 
-        oapp("--with-linalg-libs=%s" % linalg.ld_flags)
+        oapp('--with-linalg-libs={0}'.format(linalg.ld_flags))
 
         # FFTW3: use sequential or threaded version if +openmp
-        fftflavor, fftlibs = "fftw3", "-lfftw3 -lfftw3f"
+        fftflavor, fftlibs = 'fftw3', '-lfftw3 -lfftw3f'
         if '+openmp' in spec:
-            fftflavor = "fftw3-threads"
-            fftlibs = "-lfftw3_omp -lfftw3 -lfftw3f"
+            fftflavor = 'fftw3-threads'
+            fftlibs = '-lfftw3_omp -lfftw3 -lfftw3f'
 
         options.extend([
-            "--with-fft-flavor=%s" % fftflavor,
-            "--with-fft-incs=-I%s" % spec["fftw"].prefix.include,
-            "--with-fft-libs=-L%s %s" % (spec["fftw"].prefix.lib, fftlibs),
+            '--with-fft-flavor=%s' % fftflavor,
+            '--with-fft-incs=-I%s' % spec['fftw'].prefix.include,
+            '--with-fft-libs=-L%s %s' % (spec['fftw'].prefix.lib, fftlibs),
         ])
-        oapp("--with-dft-flavor=atompaw+libxc")
+        oapp('--with-dft-flavor=atompaw+libxc')
 
         # LibXC library
+        libxc = spec['libxc:fortran']
         options.extend([
-            "with_libxc_incs=-I%s" % spec["libxc"].prefix.include,
-            "with_libxc_libs=-L%s -lxcf90 -lxc" % spec["libxc"].prefix.lib,
+            'with_libxc_incs={0}'.format(libxc.cppflags),
+            'with_libxc_libs={0}'.format(libxc.libs.ld_flags + ' -lm')
         ])
 
         # Netcdf4/HDF5
-        if "+hdf5" in spec:
-            oapp("--with-trio-flavor=netcdf")
+        if '+hdf5' in spec:
+            oapp('--with-trio-flavor=netcdf')
             # Since version 8, Abinit started to use netcdf4 + hdf5 and we have
-            # to link with -lhdf5_hl -lhdf5
-            hdf_libs = "-L%s -lhdf5_hl -lhdf5" % spec["hdf5"].prefix.lib
+            # to link with the high level HDF5 library
+            hdf5 = spec['hdf5:hl']
+            netcdff = spec['netcdf-fortran:shared']
             options.extend([
-                "--with-netcdf-incs=-I%s" % (
-                    spec["netcdf-fortran"].prefix.include),
-                "--with-netcdf-libs=-L%s -lnetcdff -lnetcdf %s" % (
-                    spec["netcdf-fortran"].prefix.lib, hdf_libs),
+                '--with-netcdf-incs={0}'.format(netcdff.cppflags),
+                '--with-netcdf-libs={0}'.format(
+                    netcdff.libs.ld_flags + ' ' + hdf5.libs.ld_flags
+                ),
             ])
         else:
             # In Spack we do our best to avoid building any internally provided
             # dependencies, such as netcdf3 in this case.
-            oapp("--with-trio-flavor=none")
-
-        configure(*options)
-        make()
+            oapp('--with-trio-flavor=none')
 
-        # make("check")
-        # make("tests_in")
```

Comment:
> I don't mind waiting several minutes if it means that my installation is bug-free

I agree. Besides the `run-tests` option is also handy for my developments because I use spack to compile and tests Abinit with different libraries

My main concern is related to the fact that we use a perl scipt to compare 
files line-by-line and the test fails if the output file produced by the test
does not have the same number of lines as the reference.

One can have false negative errors because iterative algorithms may require a different number of iterations to converge (this is especially true when optimized libraries and/or aggressive optimization are used).

We have a test farm that allows us to detect tests that are not very portable and 
this allows us to tune the input parameters to make the test more stable.
Still it's quite common to have, let's say, 1-5 false alarms among the ~1000 tests and 
in this case the user should look at the output produced by `runtests.py` to understand if these errors are critical or not.